### DISCUSSION TITLE
x-ref not picking up element when id attribute set when LiveView component is present on same page

### DISCUSSION THREAD

**Discussion by mfilej** [alpinejs/alpine#2989] [7. Bugs (archived)]
It took me a while to narrow this down to the smallest example possible, but I still can't tell if it's an issue with Alpine or LiveView. I'm planning to ask on both sides. I will describe the issue first and provide a running example with code afterwards.

1. Render a live component.
2. Render an Alpine component including an element that has both `id` and `x-ref` attributes.
3. Trying to reference the element from 2) will result in `undefined`.

Things that solve the issue: a) remove `id` from element or b) render live component after Alpine component.

I used input fields in the example, but I could reproduce the issue with a span as well.

I'm providing [a running example](https://alpinejs-liveview-repro.fly.dev), [source code](https://github.com/mfilej/alpinejs-liveview-repro/blob/main/lib/repro_web/live/home_live.ex), and [the snippet](https://github.com/mfilej/alpinejs-liveview-repro/blob/main/assets/js/app.js#L40-L43) I used to initialize Alpine.

**Comment by mfilej** [comment]
I completely understand why this is difficult to debug as one has to check out the example project, install all the deps, ... Is there anything I can do to lower the barrier?

If I were to start digging deeper into alpine.js myself, what would be a good approach?

**Reply by SimoTod** [reply]
The majority of people that usually reply are php developers. You can probably avoid using $refs for now and just use something like `$root.querySelector('#YOUR-ID')`.
To try to understand better, does liveview do anything special to element with IDs? Like cloning them, detaching and reattaching it, etc.

**Reply by SimoTod** [reply]
out of curiosity, what happens if you add `phx-update="ignore"` to that input (I only had a quick look at the documentation, I hope it applies to your version)? If my gut feeling is correct, LiveView is detaching and reattaching the element and somehow Alpine is not liking it and it removes the reference from his cache

**Reply by mfilej** [reply]
@SimoTod I appreciate your response! I haven't had the chance to try this out yet, so for now, I just wanted to say thanks. Will report back when I give it a try.
